src/rate-limit.ts

```diff
@@ -0,0 +1,43 @@
+import LRU from 'lru-cache'
+import { NextApiResponse } from 'next'
+
+const MAX_UNIQUE_USERS_PER_INTERVAL = 600 // How many different users are allowed per interval
+const MAX_LIMIT_PER_USER = 3 // How many user requests are allowed per interval
+export const INTERVAL_MS = 60 * 1000 // Interval in ms
+
+export const TOKEN_CACHE_FULL = 'Endpoint is busy. Please try again later.'
+export const TOO_MANY_USER_REQUESTS = 'You have reached maximum allowed requests for now. Please, try again later.'
+
+// Inspired by https://github.com/vercel/next.js/tree/canary/examples/api-routes-rate-limit
+export const rateLimit = () => {
+  const tokenCache = new LRU<string, number[]>({
+    max: MAX_UNIQUE_USERS_PER_INTERVAL,
+    maxAge: INTERVAL_MS,
+  })
+
+  return {
+    check: (res: NextApiResponse, token: string) =>
+      new Promise((resolve, reject) => {
+        if (!tokenCache.has(token)) {
+          tokenCache.prune() // Our cache is small so we can safely prune old entries here
+
+          if (tokenCache.itemCount == MAX_UNIQUE_USERS_PER_INTERVAL) {
```

Comment:
Don't know how important this is though, but seems slightly safer
```suggestion
          if (tokenCache.itemCount >= MAX_UNIQUE_USERS_PER_INTERVAL) {
```

Reply:
Otherwise, just as a general rule, I would avoid `==`
```suggestion
          if (tokenCache.itemCount === MAX_UNIQUE_USERS_PER_INTERVAL) {
```